src/expr/src/expr/expr_array_length.rs

```diff
@@ -0,0 +1,278 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use risingwave_common::array::{ArrayRef, DataChunk};
+use risingwave_common::row::OwnedRow;
+use risingwave_common::types::{DataType, Datum, DatumRef, ScalarImpl, ScalarRefImpl, ToDatumRef};
+use risingwave_common::util::iter_util::ZipEqFast;
+use risingwave_pb::expr::expr_node::{RexNode, Type};
+use risingwave_pb::expr::ExprNode;
+
+use crate::expr::{build_from_prost, BoxedExpression, Expression};
+use crate::{bail, ensure, ExprError, Result};
+
+/// Returns the length of an array.
+///
+/// ```sql
+/// array_length ( array anyarray) → int64
```

Comment:
This looks OK to me, BUT it's different from Postgres's `array_length`

```
array_length(anyarray, int)
returns the length of the requested array dimension
```

https://www.postgresql.org/docs/8.4/functions-array.html

However, I think Postgres has this behavior because it's multi-dimentional array, but it's not in RisingWave (see https://github.com/risingwavelabs/risingwave/issues/3811).

Should we decide to diverge from PG in array's behavior now? cc @xiangjinwu 1. Should we decide to diverge from PG in array's behavior now? cc @xiangjinwu @lmatz 😇

Reply:
Oh I found @lmatz already mentioned the difference here https://github.com/risingwavelabs/risingwave/issues/7117#issuecomment-1469197802